### ISSUE TITLE
Relative paths

### ISSUE THREAD

**Issue by pnorman** [mapbox/mapbox-studio-classic#983]
I'm uncertain how to create a style using a relative path to a tm2source.

I have my style in `/home/pnorman/foo/bar.tm2` and source in `/home/pnorman/foo/bar.tm2source`. `/home/pnorman/foo` is a git repository.
If I define my source as `tmsource:///home/pnorman/mq-carto-style-vector/mq-carto-style.tm2source` it works, but this is non portable. 

I tried both `./mq-carto-style.tm2source` and `../mq-carto-style.tm2source`, but both were rejected by the UI.

What's the way to specify a relative location?

Cross-ref [RFC 3986 4.2](https://tools.ietf.org/html/rfc3986#section-4.2)

**Comment by springmeyer** [comment]
Yes, this would be quite useful. If you have ideas for how it could be done please comment more. I've personally not looked closely yet at what breaks/what would need fixing to support relative paths, but this is on my radar to research.

**Comment by pnorman** [comment]
A bare mapbox id would need to be changed to a mapbox URL. A bare id would not be suggested for portability with software other than mapbox studio (e.g. kosmtik), and having the field either able to accept a generic piece of text or a URI is confusing.

Aside from that, there's probably a node URI parser which can identify relative paths. I'm not sure if the relative paths should be relative to the project MML (project.yml) or the .tm2 directory. Not all consumers have the foo.tm2/project.yml assumption hard-coded.

cc @yohanboniface

**Comment by yhahn** [comment]
@pnorman could you describe your need rather than a particular technical solution?

Early in development we had a relative path system that we scrapped because it was brittle and asking all users using the projects involved to layout their filesystem in a similar way did not work well. We found it better to materialize the source and share it that way as a tilejson URL or mapbox map ID.

**Comment by pnorman** [comment]
I need to share a project with both a style and source with git (e.g. [natural-earth-tm2](https://github.com/mapbox/natural-earth-tm2)), where I know the path of the source relative to the style, but don't know the absolute path. I don't have a tilejson url, since I don't have a tile directory. I don't have a mapbox map ID, since it's a local source.

**Comment by alexrollin** [comment]
I am interested in this also.

## 

Alex

**Comment by four43** [comment]
This would be a really useful feature for us as well. It shouldn't be too hard to resolve something like `tmsource://../../my-source.tm2source` from the current script right? Or am I missing the complexity in this?

**Comment by four43** [comment]
@springmeyer - Is this related to mapbox studio or is this more of a mapnik core feature?
